apps/core/src/components/affine/auth/use-captcha.tsx

```diff
@@ -0,0 +1,99 @@
+import { fetchWithTraceReport } from '@affine/graphql';
+import { Turnstile } from '@marsidev/react-turnstile';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import useSWR from 'swr';
+
+import * as style from './style.css';
+
+const useCloudflareCaptcha = (): [string | null, () => JSX.Element] => {
+  const [verifyToken, setVerifyToken] = useState<string | null>(null);
+
+  const Captcha = useCallback(() => {
```

Comment:
might not be a good idea to define new component in render. Same for https://github.com/toeverything/AFFiNE/pull/4582/files#diff-f3a58898efe08b35010864aa9feea0672027e7fba87edf3791111211acfc7d5fR87